src/v/kafka/client/consumer.cc

```diff
@@ -61,6 +69,17 @@ struct partition_comp {
     }
 };
 
+fetch_response
+reduce_fetch_response(fetch_response result, fetch_response val) {
```

Comment:
take in an rvalue for retch_response or are these move-only types?

Reply:
`fetch_response` is move-only. Happy to make it an r-value ref and move it out.

Reply:
move only types is good. wasn't obvious as a reviewer that they were move-only types.